### ISSUE TITLE
Plex Trakt Configuration "unknown error"

### ISSUE THREAD

**Issue by Blessedwalnut** [trakt/Plex-Trakt-Scrobbler#323]
I have installed the plugin for Plex on the mac and can see the plugin (change settings, etc) on the channels list on Plex (using chrome)

I have a trakt account, but it says that authentication has failed.

I have tried going through http://trakt-for-plex.github.io/configuration/#/connect but when I select my server it just says "unknown error". Is there a way to fix this?

**Comment by fuzeman** [comment]
Are you using the latest version of the plugin ([v1.0.2.1](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.2.1))? this can be checked in `Channels -> Trakt.tv -> About`.

Uploading the latest plugin log file will help me identify the issue, see [this page](https://github.com/trakt/Plex-Trakt-Scrobbler/wiki/Log-files) for the location.

**Comment by Blessedwalnut** [comment]
Juste checked, running version 1.0.2.1. I've attached my latest log as a text file. Hope this helps. Sorry about the duplicate, not sure how I managed that.
[com.plexapp.plugins.trakttv copy.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/292095/com.plexapp.plugins.trakttv.copy.txt)

**Comment by fuzeman** [comment]
This seems to be the root cause of your issues with the plugin:
```
Exception raised while handling request 'system.authenticate': '_socketobject' object has no attribute 'set_tlsext_host_name'
Traceback (most recent call last):
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/plugin/api/core/manager.py", line 76, in process
    result = cls.call(method, headers, body, func, args, kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/plugin/api/core/manager.py", line 94, in call
    return func(*args, **kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/plugin/api/core/base.py", line 47, in inner
    return func(self, *args, **kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/plugin/api/system.py", line 32, in authenticate
    account = self._get_account(plex_token)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/plugin/api/system.py", line 106, in _get_account
    'X-Plex-Token': plex_token
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/api.py", line 67, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/api.py", line 53, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/sessions.py", line 468, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/adapters.py", line 376, in send
    timeout=timeout
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/packages/urllib3/connectionpool.py", line 559, in urlopen
    body=body, headers=headers)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/packages/urllib3/connectionpool.py", line 345, in _make_request
    self._validate_conn(conn)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/packages/urllib3/connectionpool.py", line 784, in _validate_conn
    conn.connect()
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/packages/urllib3/connection.py", line 252, in connect
    ssl_version=resolved_ssl_version)
  File "/Users/Nick/Library/Application Support/Plex Media Server/Plug-ins/Trakttv.bundle/Contents/Libraries/Shared/requests/packages/urllib3/contrib/pyopenssl.py", line 296, in ssl_wrap_socket
    cnx.set_tlsext_host_name(server_hostname)
AttributeError: '_socketobject' object has no attribute 'set_tlsext_host_name'
```

Will need to do some research on this issue, will let you know when I find a solution.

**Comment by Blessedwalnut** [comment]
Hi, did you have any luck with this?

**Comment by fuzeman** [comment]
Try this version of the plugin: https://github.com/trakt/Plex-Trakt-Scrobbler/archive/develop.zip

**Note:** Haven't had time to try reproduce this issue on OS X, so I'm not really sure if these changes will work or not. Please post the plugin log file again if it still doesn't work.

--------------------

> Server version is 0.9.12.8.1362-4601e39

Looks like your installation of Plex Media Server is a bit out of date, updating to the latest version (0.9.16.6.1993-5089475) might have some impact on this issue as well.

**Comment by fuzeman** [comment]
Changes released in [v1.0.3.1](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.3.1), please let me know if this issue is still happening.

**Comment by Blessedwalnut** [comment]
Thanks, this is resolved now
